common/src/test/java/org/conscrypt/TrustManagerImplTest.java

```diff
@@ -135,76 +134,32 @@ public void testHttpsEndpointIdentification() throws Exception {
         String goodHostname = TestKeyStore.CERT_HOSTNAME;
         String badHostname = "definitelywrong.nopenopenope";
 
-        // The default hostname verifier on OpenJDK rejects all hostnames, so use our own
-        javax.net.ssl.HostnameVerifier oldDefault = HttpsURLConnection.getDefaultHostnameVerifier();
         try {
-            HttpsURLConnection.setDefaultHostnameVerifier(new TestHostnameVerifier());
-
             SSLParameters params = new SSLParameters();
 
             // Without endpoint identification this should pass despite the mismatched hostname
             params.setEndpointIdentificationAlgorithm(null);
 
             List<X509Certificate> certs = tmi.getTrustedChainForServer(chain, "RSA",
-                new FakeSSLSocket(new FakeSSLSession(badHostname, chain), params));
+                    new FakeSSLSocket(new FakeSSLSession(badHostname, chain), params));
             assertEquals(Arrays.asList(chain), certs);
 
             // Turn on endpoint identification
             params.setEndpointIdentificationAlgorithm("HTTPS");
 
-            try {
-                tmi.getTrustedChainForServer(chain, "RSA",
-                    new FakeSSLSocket(new FakeSSLSession(badHostname, chain), params));
+            try { // this should fail
+                certs = tmi.getTrustedChainForServer(chain, "RSA",
+                        new FakeSSLSocket(new FakeSSLSession(badHostname, chain), params));
+                assertEquals(Arrays.asList(chain), certs);
                 fail();
-            } catch (CertificateException expected) {
-            }
-
-            certs = tmi.getTrustedChainForServer(chain, "RSA",
```

Comment:
I'm unsure why you removed these checks, they should pass on OpenJDK (you'll need to change the method signature in the anonymous class below though to match the change we're making to ConscryptHostnameVerifier).

Reply:
So I thought I was supposed to remove the code for where we set the default manually since now it does it for us. Besides that the line ```certs = tmi.getTrustedChainForServer(chain, "RSA",
                        new FakeSSLSocket(new FakeSSLSession(badHostname, chain), params));``` Is the same each time with a different default so wouldn't we now only need to test it once with the Platform specified Verifier?